File: cyanBat/src/main/java/at/msmiech/cyanbat/objects/gameobjects/Background.java
package at.msmiech.cyanbat.objects.gameobjects;

import java.util.List;

import android.graphics.Rect;
import android.util.Log;
import at.grueneis.game.framework.Input.TouchEvent;
import at.grueneis.game.framework.Graphics;
import at.grueneis.game.framework.Pixmap;
import at.msmiech.cyanbat.CyanBatGame;
import at.msmiech.cyanbat.interfaces.GameObject;
import at.msmiech.cyanbat.screens.CyanBatBaseScreen;

public class Background extends PixmapGameObject {
	public static int count = 0;
	private List<GameObject> gameObjects;

	public Background(int x, int y, Pixmap pm, List<GameObject> gameObjects) {
		super(new Rect(x, y, x + pm.getWidth(), y + pm.getHeight()), pm);
		this.gameObjects = gameObjects;
		velocity.x = -2;
	}

	@Override
	public void update(float deltaTime, List<TouchEvent> touchEvents) {
		if (DEBUG)
			Log.d(TAG, "updateBackground");
		Background.count += 1;
		if (Background.count < 2) {
			int bgArea = rect.right;
			if (bgArea - 5 < CyanBatBaseScreen.DISPLAY_HEIGHT) {
				gameObjects.add(0, new Background(
						CyanBatBaseScreen.DISPLAY_HEIGHT, 0,
						CyanBatGame.background, gameObjects));
			}
		}
		super.update(deltaTime, touchEvents);
	}

	@Override
	public void draw(Graphics g) {
		if (DEBUG)
			Log.d(TAG, "drawBackground");
		g.drawPixmap(pixmap, rect.left, rect.top);
	}
}

File: cyanBat/src/main/java/at/smiech/cyanbat/gameobjects/impl/Background.kt
package at.smiech.cyanbat.gameobjects.impl

import android.graphics.Rect
import android.util.Log

import at.grueneis.game.framework.Input.TouchEvent
import at.grueneis.game.framework.Graphics
import at.grueneis.game.framework.Pixmap
import at.smiech.cyanbat.activities.CyanBatGameActivity
import at.smiech.cyanbat.gameobjects.GameObject
import at.smiech.cyanbat.gameobjects.PixmapGameObject
import at.smiech.cyanbat.screens.CyanBatBaseScreen

class Background(x: Int, y: Int, pm: Pixmap, gameObjects: MutableList<GameObject>) : PixmapGameObject(Rect(x, y, x + pm.width, y + pm.height), pm) {
    private val gameObjects: MutableList<GameObject> = gameObjects

    init {
        velocity.x = -2f
    }

    override fun update(deltaTime: Float, touchEvents: List<TouchEvent>) {
        if (GameObject.DEBUG)
            Log.d(GameObject.TAG, "updateBackground")
        Background.count += 1
        if (Background.count < 2) {
            val bgArea = rectangle.right
            if (bgArea - 5 < CyanBatBaseScreen.DISPLAY_HEIGHT) {
                gameObjects.add(0, Background(
                        CyanBatBaseScreen.DISPLAY_HEIGHT, 0,
                        CyanBatGameActivity.background, gameObjects))
            }
        }
        super.update(deltaTime, touchEvents)
    }

    override fun draw(g: Graphics) {
        if (GameObject.DEBUG)
            Log.d(GameObject.TAG, "drawBackground")
        g.drawPixmap(pixmap, rectangle.left, rectangle.top)
    }

    companion object {
        var count = 0
    }
}
